### PR TITLE
Fix build errors by explicitly adding tsd files in the project

### LINKED ISSUE #176: Tsd reinstall often fails with 'Github rate exceeded'
This build, for instance: https://travis-ci.org/mgechev/angular2-seed/jobs/86486664#L156

Fails with

```
tsd reinstall --clean && tsd link && tsd rebundle && gulp postinstall -> running reinstall
    [ERR!] GitHub API: GitHub rate limit reached.
    To increase the limit use GitHub authentication.
    See: https://github.com/DefinitelyTyped/tsd#tsdrc
```

Documentation itself suggests to add a `token` into `.tsdrc`. Which is kind-of weird to have on Travis/CircleCI since it's a private github key.

Imagine if npm dependencies were pulled from Github. A CI provider would easily spend the 60 api requests per hour. Forgive my lack of knowledge, but why is tsd pulling this from Github and precicely what is it pulling from Github?

Is there a way that stuff inside the `tsd` command can be cached? Or pulled from somewhere else?

### PR DESCRIPTION
**Pull request by jesperronn**
Fixes #176

The pull request #173 introduced a lot of builds failing in errors because Tsd task would attempt to retrieve typings files from the https://github.com/borisyankov/DefinitelyTyped repository.

The problem is that CI builds have to use the network to retrieve the typings files, and this fails often (see discussion in #176).

https://github.com/DefinitelyTyped/tsd/#notes notes that

> It is recommended you check-in the definitions you install into your VCS.

The commit here adds all the typings files to this repository. Which is recommended to do. This is a tradeoff, since the "tsd" task must run at any point in time where we update dependencies or change type definitions.

On the other hand, build tasks can run without using the network.

The final point I am trying to make is that I would like to avoid the flakyness, and introduce as much stability in the building of the seed project as possible.

For discussion of the pros and cons, see

https://github.com/mgechev/angular2-seed/pull/173
https://github.com/mgechev/angular2-seed/issues/176
https://github.com/mgechev/angular2-seed/issues/169
https://github.com/mgechev/angular2-seed/issues/171
https://github.com/mgechev/angular2-seed/issues/184

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,12 +12,11 @@
     "karma": "karma",
     "karma.start": "karma start",
     "lint": "gulp tslint",
-    "postinstall": "tsd reinstall --clean && tsd link && tsd rebundle && gulp postinstall",
     "start": "gulp serve --env dev",
     "serve.dev": "gulp serve --env dev",
     "tasks.list": "gulp --tasks-simple",
     "test": "gulp test",
-    "tsd": "tsd"
+    "tsd": "gulp tsd"
```
Comment (ludohenin) [inline package.json]: I disagree with this as `npm run tsd -- <args>` is just a proxy to `tsd` bin for those who don't want to have it installed globally.

**Comment by ludohenin** [comment]
:+1: (except my comment)

File: package.json (hunk 1 above)
Comment (jesperronn) [inline package.json]: @ludohenin OK, I understand. Beware, though that `gulp tsd` is a proxy for the previous `tsd reinstall --clean && tsd link && tsd rebundle`. 

Should I create another task (for easy discovery)? If so, what name? 

Suggestion: `npm run tsd:reinstall` or `npm run tsd:update`

**Comment by jesperronn** [comment]
@ludohenin rewrite done. Wait for tests be finished.

**Comment by ludohenin** [comment]
I think we should close that one. I was in favor of it but I've to admit it's the cleanest way to do things.
@mgechev any thoughts ?

**Comment by mgechev** [comment]
Agree with you. I'm not a big fan of tsd either but I don't see better options.

**Comment by jesperronn** [comment]
@ludohenin @mgechev I accept your opinions for this, although I still see the problems for the build failures all over. 

You are already aware that Travis will not resolve the env parameters for pull requests, due to security reasons. This means that all pull-requests will still fail as long as tsd files are not in the repos. 

This in turn makes it harder for you to consider external contributions. 

Again, it's your project, your call.

**Comment by tarlepp** [comment]
Actually Travis should follow the ENV variable that is set it from Travis Setting GUI. I have set that from there and no problems at all after that. And that ENV variable is used when PR is made and Travis starts to run those tests.

**Pull request closed without merging**